lib/instrumenter/collectors-vm/link.js

```diff
@@ -45,13 +46,64 @@ var LINK_KSTATS = [
     }
 ];
 
+/*
+ * Because this is a latency-sensitive code path, we can't pay the
+ * price of forking a new process to run `vmadm get <:instance_uuid>`.
+ * Instead, we invoke vminfod local HTTP API --  which has not been
+ * stabilized yet -- to retrieve the instance information.
+ */
+
+var VMINFOD_TIMEOUT_MS = 500;
+
+function getInstanceNicsInfo(instanceUuid, cb) {
+
+    var timeout = {};
+    function done(err, data) {
+        if (!timeout.done) {
+            cb(err, data);
+            timeout.done = true;
+        }
+    }
+
+    setTimeout(function _to() {
+        done(new Error('Timed out waiting for vminfod'));
+    }, VMINFOD_TIMEOUT_MS);
+
+    mod_http.get('http://127.0.0.1:9090/vms/' + instanceUuid,
+        function _gotHttpResp(res) {
+
+        if (res.statusCode !== 200) {
+            res.resume();    // Consume res stream
+            done(new Error('Failed to retrieve instance info from vminfod ' +
+                'Status Code: ' + res.statusCode));
+            return;
+        }
+
+        var vmInfo = '';
+        res.on('data', function _onData(chunk) {
+            vmInfo += chunk;
+        });
+        res.on('end', function _onEnd() {
+            try {
+                var nics = JSON.parse(vmInfo).nics || [];
+                done(null, nics);
+            } catch (_e) {
+                done(new Error('Invalid payload received from vminfod: ' +
+                    vmInfo));
+            }
+        });
+    }).on('error', function _onErr(err) {
+        done(err);
+    });
+}
```

Comment:
I think you can avoid most of this code by using the restify (already included in the package.json) [JsonClient](http://restify.com/docs/client-guide/).

Reply:
Done.